sdk-lib/src/main/java/cash/z/ecc/android/sdk/jni/RustBackend.kt

```diff
@@ -341,6 +341,7 @@ internal class RustBackend private constructor(
             birthdayHeight: BlockHeight
         ): RustBackend {
             rustLibraryLoader.load()
+            initLogs()
```

Comment:
Is there a problem if `initLogs` gets called more than once?

A debug/release condition may not be sufficient, as we run benchmarks against release builds of the app.  This is because debug builds have the just in time compiler disabled, which means that performance isn't reflective of what users would see.  We do have a way to differentiate between release for benchmarking versus release for deplyoment, but that would require a bit more work to set up.

For now, what if we consider this to only enable error logging plus tracing for release builds?  Then we can circle back to figure out what debug logging might look like?

Reply:
> Is there a problem if `initLogs` gets called more than once?

Yes, calling `initLogs()` more than once will crash the application (because the global logging system can only be set up once). This is why it should be called as the Rust library is loaded, which I thought only happens once per app run, but in fact just changing screens in the demo app triggers a reload of the library for some reason. I don't actually know if this is _safe_ (that is, I don't know what the Kotlin side is doing when the library loading logic is called repeatedly).

If `initLogs()` is going to get called more than once as a matter of course, then I can insulate it on the Rust side, but it will still only set up the logging system once (it would just ignore subsequent calls).

> For now, what if we consider this to only enable error logging plus tracing for release builds? Then we can circle back to figure out what debug logging might look like?

This is what I was suggesting above: have `initLogs` enable error logging plus tracing for release builds, and have it enable debug logging plus tracing for debug builds. All that requires is a boolean flag across the JNI. If that flag is too hard though, then we can just change `initLogs` to unilaterally enable the former for all builds.

Reply:
Aha, I've looked at how `NativeLibraryLoader` works, and it does indeed load the Rust library atomically, but separately from `initLogs`. I'll modify it to take an `onLoad` callback so we can have load-and-init as a singleton.

Reply:
For simplicity, let’s just have initLogs() enable error logging and tracing for release builds.

An alternative would be to give the initLogs() method a Boolean parameter `isDebug` so that the caller can control whether the logging is for debug or release.  We’d initially hard-code isDebug = false.  I still have to implement some other changes to make toggling this from debug to release work robustly.

Reply:
> For simplicity, let’s just have initLogs() enable error logging and tracing for release builds.

This is what I have effectively implemented in fd5be1ef4e77a295f16eb22edddf9d86a93fce90 (it enables `info` logging because that is what we need for the tracing parts, but not `debug` logging).

> An alternative would be to give the initLogs() method a Boolean parameter isDebug so that the caller can control whether the logging is for debug or release. We’d initially hard-code isDebug = false.

I can do this if you want; LMK.